python/lsst/summit/utils/tmaUtils.py

```diff
@@ -150,11 +165,88 @@ def getAzimuthElevationDataForEvent(client, event, prePadding=0, postPadding=0):
+    elValues = elevationData['actualPosition'].values
+    azDemand = azimuthData['demandPosition'].values
+    elDemand = elevationData['demandPosition'].values
+
+    azError = (azValues - azDemand) * 3600
+    elError = (elValues - elDemand) * 3600
+
+    azimuthData['azError'] = azError
+    elevationData['elError'] = elError
+
     return azimuthData, elevationData
 
 
-def plotEvent(client, event, fig=None, prePadding=0, postPadding=0, commands={},
-              azimuthData=None, elevationData=None):
+def filterBadValues(values, maxDelta=0.1):
+    """Filter out bad values from a dataset, replacing them in-place.
+
+    This function replaces non-physical points in the dataset with an
+    extrapolation of the preceding two values. No more than 3 successive data
+    points are allowed to be replaced. Minimum length of the input is 3 points.
+
+    Parameters
+    ----------
+    values : `list` or `np.ndarray`
+        The dataset containing the values to be filtered.
+    maxDelta : `float`, optional
+        The maximum allowed difference between consecutive values. Values with
+        a difference greater than `maxDelta` will be considered as bad values
+        and replaced with an extrapolation.
+
+    Returns
+    -------
+    nBadPoints : `int`
+        The number of bad values that were replaced out.
+    """
+    # Find non-physical points and replace with extrapolation. No more than 3
+    # successive data points can be replaced.
+    badCounter = 0
+    consecutiveCounter = 0
+
+    log = logging.getLogger(__name__)
+
+    median = np.nanmedian(values)
+    # if either of the the first two points are more than maxDelta away from
+    # the median, replace them with the median
+    for i in range(2):
+        if abs(values[i] - median) > maxDelta:
+            values[i] = median
+            badCounter += 1
+            log.warning(f"Replacing bad value at index {i} with median value")
+
+    # from the second element of the array, walk through and calculate the
+    # difference between each element and the previous one. If the difference
+    # is greater than maxDelta, replace the element with the average of the
+    # previous two elements from the point where we started replacing.
+    replacementValue = (values[1] + values[0]) / 2.0  # in case we have to replace the first value
+    for i in range(2, len(values)):
+        if abs(values[i] - values[i-1]) > maxDelta:
+            if consecutiveCounter < 3:
+                consecutiveCounter += 1
+                badCounter += 1
+                log.warning(f"Replacing value at index {i} with {replacementValue}")
+                values[i] = replacementValue
+            else:
+                log.warning(f"More than 3 consecutive replacements at index {i}. Stopping replacements"
+                            " until the next good value.")
+        else:
+            consecutiveCounter = 0
+            replacementValue = (values[1] + values[0]) / 2.0
```

Comment:
The comments lead me to believe this would be `(values[i] + values[i-1])/2.0`... this is just the average of the first two elements instead? Which you calculated outside of the loop and should not have changed.

But note even if it were the average of `values[i]` and `values[i-1]`, it's possible that `values[i-1]` was itself a bad replaced value, or a bad value that wasn't replaced at all if `consecutiveCounter` was too large.

Reply:
This was a good point. I'd meant to deal with this issue before, but didn't because the effort/reward seemed poor given where this goes in the codebase, but I ended up agreeing with you and putting it in, along with tests.